lib/softio/softio_spi.h

```diff
@@ -0,0 +1,160 @@
+#pragma once
+
+/**
+ * @file softio_spi.h
+ * Software (bit-banged) SPI implementation. Master-only. Supports all 4 modes
+ * with clock rates of up to 200kHz.
+ * 
+ * @example
+ * ```c
+ * // initialize
+ * SoftIoSpiBusConfig bus_cfg = {
+ *     .miso = &gpio_ext_pa6,
+ *     .mosi = &gpio_ext_pa7,
+ *     .sck = &gpio_ext_pb3,
+ *     .clk_polarity = 0,
+ * };
+ * SoftIoSpiSlaveConfig dev_cfg = {
```

Comment:
Bus slaves looks kinda like overkill, also there are cases when you'd want to manage CS line outside of the spi scope. For example cc1101 driver is doing that.